src/custom_dispatch.jl

```diff
@@ -0,0 +1,108 @@
+import Core: SSAValue
+using Cassette
+
+function isinteresting end
+function alternative end
+
+# Must return 8 exprs
+function rewrite_call(ctx, stmt, extraslot, i)
+    exprs = Any[]
+    cond = stmt.args[1]        # already an SSAValue
+
+    # order of expressions we pre-assign
+    # SSA for each expression here
+
+    ISINTERESTING,
+    GOTOIFNOTINTERESTING,
+    ALTERNATIVE,
+    ALTERNATIVE_IN_SLOT,
+    GOTO_LAST,
+    ORIGINAL,
+    ORIGINAL_IN_SLOT,
+    LAST = i:i+20
+
+
+    # ISINTERESTING
+    push!(exprs, Expr(:call,
+                      Expr(:nooverdub, isinteresting),
+                      Expr(:contextslot),
+                      stmt.args...))
+
+    # GOTOIFNOTINTERESTING
+    push!(exprs, Expr(:gotoifnot, SSAValue(ISINTERESTING), ORIGINAL))
+
+    # ALTERNATIVE
+    push!(exprs, Expr(:call,
+                      Expr(:nooverdub, alternative),
+                      Expr(:contextslot),
+                      stmt.args...))
+
+    # ALTERNATIVE_IN_SLOT
+    push!(exprs, Expr(:(=), extraslot, SSAValue(ALTERNATIVE)))
+
+    # GOTO_LAST
+    push!(exprs, Core.GotoNode(LAST))
+
+    # ORIGINAL
+    push!(exprs, stmt)
+
+    # ORIGINAL_IN_SLOT
+    push!(exprs, Expr(:(=), extraslot, SSAValue(ORIGINAL)))
+
+    # LAST
```

Comment:
Comment here that this is required to be the same value as the original expression we were replacing,